workshop/content/advanced/resource-types/example-in_Python/_index.md

```diff
@@ -0,0 +1,334 @@
+
+```
+mkdir example-testing
+cd example-testing
+cfn init
+```
+
+Choose to specify `r` to develop a new resource, and proceed by pressing Return/Enter. You will be prompted to enter the name of your resource type: specify so by following the `Organization::Service::Resource` pattern; for example, specify `MyOrganization::MyService::MyResource` and choose to continue.
+
+Next, you will be prompted to specify a language to use for code generation, as in the following example output:
+
+```
+>> MyOrganization::MyService::MyResource
+Select a language for code generation:
+[1] go
+[2] java
+[3] python36
+[4] python37
+[5] typescript
+(enter an integer): 
+>>
+```
+
+Choose the line item relevant to Python: in the example above, you choose to specify `4` for `python37`.
+
+On the net question, choose to use Docker, and then continue.
+
+Next, you should get a message notifying you that your project has been initialized in the current directory.
+
+Congratulations! You have created your first stub for your example resource type in Python! Let's take a look at some of the generated stub content:
+
+* `README.md`: main documentation file where you'd want to add information on how to use your resource type;
+* `docs`: directory containing syntax information for properties of your resource type;
+* `example_inputs`: directory containing files with property key/value data you will specify for use with contract tests. *As such files are also added to source code control, do not add sensitive information to such files*;
+* `myorganization-myservice-myresource.json`: this file, named after your resource type name choice above, contains the schema for the resource: **this file is where you describe the model for your resource**;
+* `src`: contains a directory named after your resource type name, inside of which you should find the `models.py` file, that you manage with the CloudFormation CLI when you make schema changes, and the `handlers.py` file, that is where you will start adding your code for the CRUDL implementation logic. Open the `src/handlers.py` file with a text editor of your choice, and **familiarize with the handlers' structure** described in `create_handler`, `update_handler`, `delete_handler`, `read_handler`, `list_handler` functions;
+* `resource-role.yaml`: file created by the CloudFormation CLI, that describes an [AWS Identity and Access Management](https://aws.amazon.com/iam/) (IAM) role whose `PolicyDocument` contains permissions you indicate in the schema for your handlers in the `handlers` section.  CloudFormation assumes this role to manage resources on your behalf as part of CRUDL operations;
+* `template.yml`: [AWS Serverless Application Model](https://aws.amazon.com/serverless/sam/) (SAM) template used as part of resource type testing.
+
+
+
+#### Example resource type walkthrough
```

Comment:
The lab mainly focuses on thought process and example considerations to make when designing a resource model and building handler logic. It offers the existing resource as a sample to compare takeaways, and to see those applied in the sample itself. It then also provides hands-on actions on topics that include creating the infrastructure for contract tests, running unit tests, submitting to the registry and offering, in the challenge section, the action of writing a JSON file test for manually testing a handler.

Reply:
In the context of a hands-on, real-world workshop with 100 users in a room expecting to learn how to build resource types, I feel it would be good to have some step-by-step guidance on how to get started from scratch. We could include this advice in addition to providing an example resource type to work through.